### PR TITLE
Added maestro test flows

### PR DESCRIPTION
**Pull request by wmontwe**
Reviewing #6632 showed that it is hard to verify the intended behavior.

This is an attempt to add UI E2E test flows to verify that behavior and also provide some flows to cover repetitive tasks. It introduces [Maestro](https://maestro.mobile.dev/) to write flows for Android that cover interaction with other apps. In this case contacts.

To be able to run the flows, you need to [install the CLI tools](https://maestro.mobile.dev/getting-started/installing-maestro). Then ensure a device or emulator is running and just execute:

- `maestro test ui-flows/validate_compose_simple_message.yml `
- `maestro test ui-flows/validate_message_details_show_contact_names.yml`

Once maestor validates as a reliable tool, flows could also be executed as part of the CI process.

This depended on #6632 but rebased on #6642

### CODE REVIEW

**Comment by cketti** [comment]
The `validate_compose_simple_message` flow didn't run successfully for me at first. The problem was that `bob@example.com` was in my contacts. We probably need to come up with a naming convention to make sure this doesn't happen. Maybe: `bob@not-in-contacts.example`

**Comment by cketti** [comment]
The `validate_message_details_show_contact_names` flow depends on the UI layout of the contact app installed on the device. We should add a comment to list the emulator options required to make the test pass.

**Comment by wmontwe** [comment]
To reduce the UI dependency of `validate_message_details_show_contact_names` we could also remove the second contact to check. I added it as it was part of the pull-request example screenshot. But I'll add information about the used emulator anyway.

**Comment by wmontwe** [comment]
`validate_compose_simple_message` could indeed use a different name, to avoid collisions

File: ui-flows/shared/login_demo_account.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+- tapOn:
+    id: "com.fsck.k9.debug:id/account_email"
+- inputText: "demo@k9mail.example"
+- tapOn:
+    id: "com.fsck.k9.debug:id/next"
+
+## Account name configuration
+- tapOn:
+    id: "com.fsck.k9.debug:id/account_description"
+- inputText: "demo@k9mail"
```
Comment (cketti) [inline ui-flows/shared/login_demo_account.yml:20]: This is the name of the account. So this should probably be "Demo Account".

File: ui-flows/shared/login_demo_account.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+- tapOn:
+    id: "com.fsck.k9.debug:id/account_name"
+- inputText: "Demo Account"
```
Comment (cketti) [inline ui-flows/shared/login_demo_account.yml:23]: This is the name that is used in outgoing messages. So it should probably be something like "Demo User".

**Review by cketti — APPROVED** [review]

File: ui-flows/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+- `shared` - flows usable by other flows
+- `validate` - flows that assert behavior of the app
+
+## Install
+
+To be able to run the flows, you need to [install the CLI tools](https://maestro.mobile.dev/getting-started/installing-maestro)
+
+## Run
+
+The current flows run best with a Pixel 2 emulator having a 5.0" screen with 1080x1920 resolution and 420dpi.
```
[the remark this answers is not in the document]
Comment (wmontwe) [reply ui-flows/README.md:17]: Right, it works for API 31. On 33 the contact provider changed it's UI.

File: ui-flows/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+## Install
+
+To be able to run the flows, you need to [install the CLI tools](https://maestro.mobile.dev/getting-started/installing-maestro)
+
+## Run
+
+Ensure a device or emulator is running and execute:
+
+- `maestro test ui-flows/validate_compose_simple_message.yml `
+- `maestro test ui-flows/validate_message_details_show_contact_names.yml`
```
Comment (cketti) [inline ui-flows/README.md:26]:
```suggestion
- `maestro test ui-flows/validate/compose_simple_message.yml `
- `maestro test ui-flows/validate/message_details_show_contact_names.yml`
```

**Review by cketti — APPROVED** [review]
:+1:

**Pull request merged**